rmw_implementation/CMakeLists.txt

```diff
@@ -15,65 +15,50 @@ find_package(ament_cmake REQUIRED)
 
 find_package(rcutils REQUIRED)
 
-# provides FindPoco.cmake and Poco on platforms without it
-find_package(poco_vendor)
-find_package(Poco COMPONENTS Foundation)
 find_package(rmw_implementation_cmake REQUIRED)
 
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
 endif()
 
+find_package(rmw REQUIRED)
 get_default_rmw_implementation(RMW_IMPLEMENTATION)
 
-get_available_rmw_implementations(RMW_IMPLEMENTATIONS)
-message(STATUS "")
-message(STATUS "Found these rmw implementations:")
-foreach(impl IN LISTS RMW_IMPLEMENTATIONS)
-  message(STATUS "  '${impl}'")
-endforeach()
-
 message(STATUS "")
 message(STATUS "Default rmw implementation: '${RMW_IMPLEMENTATION}'")
 message(STATUS "")
 
-option(RMW_IMPLEMENTATION_FORCE_DYNAMIC_LOADING
-  "Force use of dynamic loading (via poco), even if there is only one implementation available"
-  OFF)
-
-# Nominally use dynamic loading if poco is available.
-set(RMW_IMPLEMENTATION_SUPPORTS_POCO ${Poco_FOUND})
-# Determine if we're going to actually support dynamic loading.
-if(RMW_IMPLEMENTATION_FORCE_DYNAMIC_LOADING)
-  # Force usage.
-  if(NOT Poco_FOUND)
-    message(FATAL_ERROR "Dynamic loading forced, but poco is not found!")
-  endif()
-  set(RMW_IMPLEMENTATION_SUPPORTS_POCO TRUE)
-elseif(NOT RMW_IMPLEMENTATIONS MATCHES ";")
-  set(RMW_IMPLEMENTATION_SUPPORTS_POCO FALSE)
+get_available_rmw_implementations(available_rmw_implementations)
+list(LENGTH available_rmw_implementations count_available_rmw_implementations)
+if(count_available_rmw_implementations EQUAL 1)
+  set(rmw_implementation_baked_default ON)
+else()
+  set(rmw_implementation_baked_default OFF)
 endif()
 
-if(NOT RMW_IMPLEMENTATION_SUPPORTS_POCO)
-  message(STATUS "Dynamic loading disabled; directly referencing '${RMW_IMPLEMENTATION}'")
-  message(STATUS "")
+option(RMW_IMPLEMENTATION_BAKED "\
+  Use only the RMW implementation specified at build time, \
+  ignoring runtime value of the 'RMW_IMPLEMENTATION' environment variable \
+  and the 'RMW_IMPLEMENTATION' CMake option"
+  ${rmw_implementation_baked_default})
 
+if(RMW_IMPLEMENTATION_BAKED)
+  message(STATUS "Dynamic loading disabled; directly referencing '${RMW_IMPLEMENTATION}'")
 else()
-  message(STATUS "Dynamic loading enabled with any of the available rmw implementations")
-  message(STATUS "")
+  # provides FindPoco.cmake and Poco on platforms without it
+  find_package(poco_vendor REQUIRED)
+  find_package(Poco REQUIRED COMPONENTS Foundation)
 
-  find_package(rmw REQUIRED)
+  message(STATUS "Dynamic loading enabled")
```

Comment:
A more specific message would be better since the context might not be obvious for users, e.g.:

> Dynamic loading of RMW implementation enabled

Also above for the disabled case.